pkg/volume/util.go

```diff
@@ -354,3 +354,29 @@ func UnmountViaEmptyDir(dir string, host VolumeHost, volName string, volSpec Spe
 	}
 	return wrapped.TearDownAt(dir)
 }
+
+// MountOptionFromSpec extracts and joins mount options from volume spec with supplied options
+func MountOptionFromSpec(spec *Spec, options ...string) []string {
+	if spec.PersistentVolume != nil {
+		return JoinMountOptions(spec.PersistentVolume.Spec.MountOptions, options)
+	}
+	return options
+}
+
+// JoinMountOptions joins mount options eliminating duplicates
+func JoinMountOptions(userOptions []string, systemOptions []string) []string {
+	allMountOptions := map[string]string{}
```

Comment:
Really what you need here is a set not a map. Maybe use https://github.com/kubernetes/kubernetes/tree/master/pkg/util/sets for clarity.